lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,48 +31,69 @@ const ignoredPathComponents = [
-    if (comment.range[0] > endRange) continue;
-    if (comment.value.includes('@')) {
-      // This is a complex description with description and examples.
-      let description = '';
-      /** @type {Record<string, string>} */
-      const examples = {};
-
-      const r = /@(\w+) ({.+})?(.*)(\n|$)/g;
-      let matches;
-      while ((matches = r.exec(comment.value)) !== null) {
-        const tagName = matches[1];
-        const example = matches[2];
-        const messageOrPlaceholder = matches[3].trim();
-
-        if (tagName === 'description') {
-          description = messageOrPlaceholder;
-        } else if (tagName === 'example') {
-          examples[messageOrPlaceholder] = example.substring(1, example.length - 1);
-        }
-      }
+/**
+ * Extract the description and examples (if any) from a jsDoc annotation.
+ * @param {import('typescript').JSDoc|undefined} ast
+ * @param {string} message
+ * @return {{description: string, examples: Record<string, string>}}
+ */
+function computeDescription(ast, message) {
+  if (!ast) {
+    throw Error(`Missing description comment for message "${message}"`);
+  }
 
-      // Make sure description is not empty
-      if (description.length === 0) throw Error(`Empty @description for message "${value}"`);
-      return {description, examples};
+  if (ast.tags) {
+    // This is a complex description with description and examples.
+    let description = '';
+    /** @type {Record<string, string>} */
+    const examples = {};
+
+    for (const tag of ast.tags) {
+      const comment = coerceToSingleLineAndTrim(tag.comment);
+
+      if (tag.tagName.text === 'description') {
+        description = comment;
+      } else if (tag.tagName.text === 'example') {
+        const {placeholderName, exampleValue} = parseExampleJsDoc(comment);
+        examples[placeholderName] = exampleValue;
+      } else {
+        throw new Error('bad tag' + tag.tagName.text);
+      }
     }
 
-    /** @type {string} */
-    const description = comment.value.replace('*', '').trim();
-
-    // Make sure description is not empty
-    if (description.length === 0) throw Error(`Empty description for message "${value}"`);
+    if (description.length === 0) throw Error(`Empty @description for message "${message}"`);
+    return {description, examples};
+  }
 
+  if (ast.comment) {
     // The entire comment is the description, so return everything.
-    return {description};
+    return {description: coerceToSingleLineAndTrim(ast.comment), examples: {}};
   }
-  throw Error(`No Description for message "${value}"`);
+
+  throw Error(`Missing description comment for message "${message}"`);
+}
+
+/**
+ * Collapses a jsdoc comment into a single line and trims whitespace.
+ * @param {string=} comment
+ * @return {string}
+ */
+function coerceToSingleLineAndTrim(comment = '') {
+  // Line breaks within a jsdoc comment should always be replaceable with a space.
+  return comment.replace(/\n/g, ' ').trim();
```

Comment:
FWIW I tend to go with `/\n+/g, ' '` if i'm turning into single line in other contexts, not sure if this is different

Reply:
> FWIW I tend to go with `/\n+/g, ' '` if i'm turning into single line in other contexts, not sure if this is different

it might be stretching interpreting a string with multiple line breaks in it as still the same jsdoc string but I think it's the only thing that makes sense here, so, sounds good :)